File: 18주차/이연상/숫자의 표현.py
# [PRG] 숫자의 표현

def solution(s):
    answer = [0 , 0]
    
    while(s != "1"):
        answer[1] += s.count('0')
        s = s.replace('0', "")
        s = bin(len(s))
        s = str(s[2:])
        answer[0] += 1
    
    return answer

File: 43주차/이수민/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int answer = 1;
        int s = 1;
        int e =2;
        
        int sum = 1;
        
        while(true){
            if(e>=n || s == e) break;
            
            if(sum+e<n){
                sum+= e++;
            }else if(sum+e == n){
                answer++;
                sum+=e;
                e++;
                sum-=s;
                s++;
            }else if(sum+e>n){
                sum-=s;
                s++;
            }
        }
        return answer;
    }
}
